src/CodeGen_GPU_Host.cpp

```diff
@@ -191,8 +191,8 @@ void CodeGen_GPU_Host<CodeGen_CPU>::compile_func(const LoweredFunc &f,
         std::vector<char> kernel_src = gpu_codegen->compile_to_src();
 
         Value *kernel_src_ptr =
-            CodeGen_CPU::create_constant_binary_blob(kernel_src,
-                                                     "halide_" + function_name + "_" + api_unique_name + "_kernel_src");
+            CodeGen_CPU::create_binary_blob(kernel_src,
```

Comment:
Ow ow ow, can we please not have that as an API? 
(1) given the name, it's definitely not what I would expect
(2) boolean parameters of this sort tend to be nonobvious when reading (what does this 'true' / 'false' mean?)

IMHO a purpose-built enum that is required would vastly improve readability at point of call.